src/control/jobs/Scheduler.cpp

```diff
@@ -77,38 +65,42 @@ void Scheduler::stop() {
 void Scheduler::addJob(Job* job, JobPriority priority) {
     SDEBUG("Adding job...");
 
-    g_mutex_lock(&this->jobQueueMutex);
+    {
+        std::lock_guard lock{this->jobQueueMutex};
 
-    job->ref();
-    g_queue_push_tail(this->jobQueue[priority], job);
-    g_cond_broadcast(&this->jobQueueCond);
-
-    SDEBUG("add job: %" PRId64, (uint64_t)job);
+        job->ref();
+        this->jobQueue[priority]->push_back(job);
+    }
 
-    g_mutex_unlock(&this->jobQueueMutex);
+    SDEBUG("add job: %" PRId64 "; type: %" PRId64, (uint64_t)job, (uint64_t)job->getType());
+    this->jobQueueCond.notify_all();
 }
 
 auto Scheduler::getNextJobUnlocked(bool onlyNotRender, bool* hasRenderJobs) -> Job* {
     Job* job = nullptr;
 
     for (int i = JOB_PRIORITY_URGENT; i < JOB_N_PRIORITIES; i++) {
+        std::deque<Job*>& queue = *this->jobQueue[i];
+
         if (onlyNotRender) {
-            for (GList* l = this->jobQueue[i]->head; l != nullptr; l = l->next) {
-                job = static_cast<Job*>(l->data);
+            for (auto it = queue.begin(); it != queue.end(); ++it) {
+                job = *it;
 
                 if (job->getType() != JOB_TYPE_RENDER) {
-                    g_queue_delete_link(this->jobQueue[i], l);
+                    queue.erase(it);
```

Comment:
This is a bug, the iterators are invalidated here, so `it` has to be reassigned.